proto/viam/provisioning/provisioning.proto

```diff
@@ -0,0 +1,78 @@
+syntax = "proto3";
+
+package viam.provisioning.v1;
+
+option go_package = "go.viam.com/api/provisioning/v1";
+
+// ProvisioningService is used by viam-agent to retrieve device and smart machine info.
+service ProvisioningService {
+  // GetDeviceStatus is for retrieving the status of the device including networking.
+  rpc GetDeviceStatus(GetDeviceStatusRequest) returns (GetDeviceStatusResponse);
+
+  // SetWifiCredentials is to set the wifi credentials.
+  rpc SetWifiCredentials(SetWifiCredentialsRequest) returns (SetWifiCredentialsResponse);
+
+  // SetSmartMachineCredentials is to set the smart machine credentials.
+  rpc SetSmartMachineCredentials(SetSmartMachineCredentialsRequest) returns (SetSmartMachineCredentialsResponse); 
+
+  // GetNetworkList is to retrieve the list of networks that are visible to the device.
+  rpc GetNetworkList(GetNetworkListRequest) returns (GetNetworkListResponse); 
+}
+
+message GetDeviceStatusRequest {
+}
+
+message GetDeviceStatusResponse {
+  ProviosioningInfo provisioning_info = 1;
+  bool has_device_credentials = 2;
+  bool is_online = 3;
+  NetworkInfo latest_connection_attempt = 4;
+  repeated string errors = 5;
+}
+
+message SetWifiCredentialsRequest {
```

Comment:
As we called it "GetNetworkList()" for the other RPC, this maybe should be "SetNetworkCredentials" as well to match. In which case it should have a "type" field as #1 I think. Thoughts?